x/mongo/driver/operation.go

```diff
@@ -627,9 +638,12 @@ func (op Operation) Execute(ctx context.Context) error {
 			}
 
 			// If the error is no longer retryable and has the NoWritesPerformed label, then we should
-			// return the previous indefinite error.
-			if tt.HasErrorLabel(NoWritesPerformed) {
-				return prevIndefiniteErr
+			// set the error to the "previous indefinite error" unless the current error is already the
+			// "previous indefinite error". After reseting, repeat the error check.
+			if tt.HasErrorLabel(NoWritesPerformed) && !errors.Is(tt, prevIndefiniteErr) {
```

Comment:
Do we return the current error here if it is the "previous indefinite error"?

Reply:
If `tt == prevIndefiniteErr`, then we can just let the switch condition continue with the logic as-is. I believe that will result in the current error (i.e. `tt`) being propagated to the user.